core/templates/dev/head/pages/exploration_editor/ExplorationSaveService.js

```diff
@@ -198,36 +198,51 @@ oppia.factory('explorationSaveService', [
       },
 
       discardChanges: function() {
-        var confirmDiscard = confirm(
-          'Are you sure you want to discard your changes?');
-
-        if (confirmDiscard) {
-          alertsService.clearWarnings();
-          $rootScope.$broadcast('externalSave');
+        var confirmDiscard = $modal.open({
+          templateUrl: 'modals/confirmDiscardChanges',
+          backdrop: 'static',
+          keyboard: false,
+          controller: [
+            '$scope', '$modalInstance', function($scope, $modalInstance) {
+              $scope.close = function(result) {
+                if (result) {
+                  $modalInstance.close(true);
+                } else {
+                  $modalInstance.close(false);
+                }
+              };
+            }]
+        });
 
-          $modal.open({
-            templateUrl: 'modals/reloadingEditor',
-            backdrop: 'static',
-            keyboard: false,
-            controller: [
-              '$scope', '$modalInstance', function($scope, $modalInstance) {
-                $timeout(function() {
-                  $modalInstance.dismiss('cancel');
-                }, 2500);
-              }
-            ],
-            windowClass: 'oppia-loading-modal'
-          });
+        confirmDiscard.result.then(function(result) {
```

Comment:
Just a general note: You'll probably end up getting rid of the 'result' argument, but in the future, please try to use a more semantic name that clearly indicates what the argument represents. 'result' is very generic/vague.